lang/src/org/partiql/lang/prettyprint/QueryPrettyPrinter.kt

```diff
@@ -0,0 +1,899 @@
+package org.partiql.lang.prettyprint
+
+import com.amazon.ion.system.IonSystemBuilder
+import org.partiql.lang.domains.PartiqlAst
+import org.partiql.lang.syntax.SqlParser
+import org.partiql.pig.runtime.toIonElement
+import java.lang.StringBuilder
+import java.time.LocalDate
+import java.time.LocalTime
+import kotlin.IllegalStateException
+import kotlin.math.abs
+
+/**
+ * This class is used to pretty print a query, which first transforms a query into a parsed tree,
+ * and then transform it back to a pretty string.
+ *
+ * The idea is to use a StringBuilder and write a pretty-printed query to it according to the like
+ * of the parsed tree
+ */
+class QueryPrettyPrinter {
+    private val sqlParser = SqlParser(IonSystemBuilder.standard().build())
+
+    fun prettyPrintQuery(query: String): String =
```

Comment:
Not implying this is not a useful function but I'm just wondering what a use-case for it can be that requires us to transform to AST and back, mind elaborating?

Reply:
I suppose you are talking about `fun prettyPrintQuery(query: String): String`. I think sometimes the input of a PartiQL expression can be either in format of a string or an AST node.

Reply:
That's why I exposed 2 APIs here, which are `fun prettyPrintQuery(query: String): String` and `fun prettyPrintQuery(ast: PartiqlAst.Statement): String`